src/rgw/rgw_acl.h

```diff
@@ -214,57 +217,30 @@ struct ACLReferer {
       perm(perm) {
   }
 
-  bool is_match(std::string http_referer) const {
-    std::string http_host;
-    if (!get_http_host(http_referer, http_host))
+  bool is_match(boost::string_ref http_referer) const {
+    boost::optional<boost::string_ref> http_host = get_http_host(http_referer);
+    if ((*http_host).empty()) {
       return false;
+    }
 
-    if (http_host == url_spec) {
+    boost::string_ref host_spec(url_spec);
+    if (*http_host == host_spec) {
       return true;
     }
 
-    if (http_host.length() < url_spec.length()) {
+    if ((*http_host).length() < host_spec.length()) {
       return false;
     }
 
-    if ('.' == url_spec[0]) {
+    if ('.' == host_spec[0]) {
       /* Wildcard support: a referer matches the spec when its last char are
        * perfectly equal to spec. */
-      return !http_host.compare(http_host.length() - url_spec.length(),
-                                   url_spec.length(), url_spec);
+      return (*http_host).ends_with(host_spec);
```

Comment:
Wow, now the code is really much easier to understand. I missed the opportunity for `ends_with` last time. Thanks for applying it here!